packages/next/src/views/Root/SyncClientConfig.tsx

```diff
@@ -0,0 +1,15 @@
+'use client'
+import type { ClientConfig } from 'payload'
+
+import { useConfig } from '@payloadcms/ui'
+import { useEffect } from 'react'
+
+export const SyncClientConfig = ({ clientConfig }: { clientConfig: ClientConfig }) => {
```

Comment:
Why do we need this? The ConfigProvider already has a useEffect that reacts to prop changes.
I'd definitely add JSDocs here if this is needed